gulp/util.js

```diff
@@ -223,3 +225,75 @@ function themeBuildStream() {
       .pipe(sass())
       .pipe(utils.cssToNgConstant('material.core', '$MD_THEME_CSS'));
 }
+
+// Removes duplicated CSS properties.
+function dedupeCss() {
+  var prefixes = ['-webkit-', '-moz-', '-ms-', '-o-'];
+
+  return insert.transform(function(contents) {
+    // Parse the CSS into an AST.
+    var parsed = postcss.parse(contents);
+    var removedSelectors = [];
+    var removedCount = 0;
+
+    // Walk through all the rules, skipping comments, media queries etc.
+    parsed.walk(function(rule) {
+      if (rule.type !== 'rule' || !rule.nodes || rule.nodes.length < 2) return;
+
+      // Walk all of the properties within a rule.
+      rule.walk(function(prop) {
+        // Check if there's a similar property that comes after the current one.
+        // Note that we skip the prefixed properties and values, because they should
+        // be allowed to be repeated.
+        var hasDuplicates = find(rule.nodes, function(otherProp) {
+          return otherProp !== prop &&
+            prop.prop &&
+            otherProp.prop &&
+            otherProp.prop === prop.prop &&
+            !containsPrefix(prop.prop) &&
+            !containsPrefix(prop.value) &&
+            !containsPrefix(otherProp.prop) &&
+            !containsPrefix(otherProp.value);
+        });
+
+        // Remove the declaration if it's duplicated. Also keep track of
+        // the removed ones so we can log them later.
+        if (hasDuplicates) {
+          prop.remove();
+          removedCount++;
+
+          if (removedSelectors.indexOf(rule.selector) === -1) {
+            removedSelectors.push(rule.selector);
+          }
+        }
+      });
+    });
+
+    if (removedCount) {
+      gutil.log(
+        gutil.colors.yellow('Removed ' + removedCount + ' duplicate CSS properties from selectors: ' +
+          removedSelectors.map(function(selector) {
+            return '"' + selector + '"';
+          }).join(', ')
+        )
+      );
+    }
+
+    // Turn the AST back into CSS.
+    return parsed.toResult().css;
+  });
+
+  function containsPrefix(str) {
+    return str && str.indexOf('-') > -1 && !!find(prefixes, function(prefix) {
+      return str.indexOf(prefix) > -1;
+    });
```

Comment:
It's actually faster too, after I replaced it, the whole method went from ~120ms to 30-50ms. Although I had to tweak it since it didn't pick up the prefixed transitions properly.